msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java

```diff
@@ -188,4 +197,49 @@ private AcquireTokenResult getAcquireTokenResult(final Bundle resultBundle) thro
         throw resultAdapter.baseExceptionFromBundle(resultBundle);
     }
 
+    /**
+     * Checks if the account returns is a MSA Account and sets single on state in cache
+     * @param resultBundle
+     * @param msalOAuth2TokenCache
+     */
+    private void saveMsaAccountToCache(@NonNull final Bundle resultBundle,
+                                       @NonNull final MsalOAuth2TokenCache msalOAuth2TokenCache) {
+        final BrokerResult brokerResult = (BrokerResult) resultBundle.getSerializable(
+                AuthenticationConstants.Broker.BROKER_RESULT_V2
+        );
+
+        if (resultBundle.getBoolean(AuthenticationConstants.Broker.BROKER_REQUEST_V2_SUCCESS)
+                && brokerResult != null &&
+                AzureActiveDirectoryAudience.MSA_MEGA_TENANT_ID.equalsIgnoreCase(brokerResult.getTenantId())) {
+            Logger.info(TAG, "Result returned for MSA Account, saving to cache");
+
+            try {
+                final ClientInfo clientInfo = new ClientInfo(brokerResult.getClientInfo());
+                final MicrosoftStsAccount microsoftStsAccount = new MicrosoftStsAccount(
+                        new IDToken(brokerResult.getIdToken()),
+                        clientInfo
+                );
+                microsoftStsAccount.setEnvironment(brokerResult.getEnvironment());
+
+                final MicrosoftRefreshToken microsoftRefreshToken = new MicrosoftRefreshToken(
+                        brokerResult.getRefreshToken(),
+                        clientInfo,
+                        brokerResult.getScope(),
+                        brokerResult.getClientId(),
+                        false, // TODO : is family refresh token valid here ? Should we get this info from Broker
+                        brokerResult.getEnvironment(),
+                        null
+                );
+
+                msalOAuth2TokenCache.setSingleSignOnState(microsoftStsAccount, microsoftRefreshToken);
+
+            } catch (Exception e) {
+                Logger.error(TAG, "Exception while creating Idtoken or ClientInfo," +
```

Comment:
Let's also do a `Logger.errorPII()` on the `Exception` (or subtype), if it's thrown- the likelihood of the `ServiceException` being thrown by the `IDToken` constructor is low, given that the broker would have had to parse it to make a MSA/non-MSA determination in the first place